grpc_server_test.go

```diff
@@ -110,8 +110,8 @@ func TestGetByIPTinkerbell(t *testing.T) {
 				hw.ID, test.id)
 		}
 
-		if reflect.DeepEqual(&hw.Metadata, &packet.Metadata{}) { // return if metadata is empty
-			return
+		if reflect.DeepEqual(&hw.Metadata, &packet.Metadata{}) { // continue if metadata is empty
```

Comment:
I think this `If` condition is not required. Are you keeping it for some future use?

Reply:
It's required to have a mix of test cases where some hardware data may have metadata and others may not. If we remove this and the hardware data does not metadata, it will error out at line 116.